src/test/java/io/reactivex/observers/TestObserverTest.java

```diff
@@ -1383,6 +1383,48 @@ public void assertValueAtInvalidIndex() {
     }
 
     @Test
+    public void assertValueAtIndexEmpty() {
+        TestObserver<Object> ts = new TestObserver<Object>();
+
+        Observable.empty().subscribe(ts);
+
+        thrown.expect(AssertionError.class);
+        thrown.expectMessage("No values");
+        ts.assertValueAt(0, 1);
+    }
+
+    @Test
+    public void assertValueAtIndexMatch() {
+        TestObserver<Integer> ts = new TestObserver<Integer>();
+
+        Observable.just(1, 2).subscribe(ts);
+
+        ts.assertValueAt(1, 2);
+    }
+
+    @Test
+    public void assertValueAtIndexNoMatch() {
+        TestObserver<Integer> ts = new TestObserver<Integer>();
+
+        Observable.just(1, 2, 3).subscribe(ts);
+
+        thrown.expect(AssertionError.class);
+        thrown.expectMessage("Expected: 2 (class: Integer), Actual: 3 (class: Integer) (latch = 0, values = 3, errors = 0, completions = 1)");
+        ts.assertValueAt(2, 2);
```

Comment:
Can you please compare something else than integers (String will be ok)? 

Silly code change in assert method can break its logic but pass this test because both index and value are `2` :)